subsys/spm/spm.c

```diff
@@ -472,33 +467,29 @@ static void spm_config_peripherals(void)
 
 	};
 
-	if (IS_ENABLED(CONFIG_SPM_NRF_DPPIC_NS)) {
-		spm_dppi_configure(CONFIG_SPM_NRF_DPPIC_PERM_MASK);
-	}
+#if defined(CONFIG_NRF_DPPI_CHANNEL_MASK_SECURE)
+	NRF_SPU->DPPI[0].PERM = CONFIG_NRF_DPPI_CHANNEL_MASK_SECURE;
+#endif
 
 	PRINT("Peripheral\t\tDomain\t\tStatus\n");
 
-	if (IS_ENABLED(CONFIG_SPM_NRF_P0_NS)) {
-		/* Configure GPIO pins to be Non-Secure */
-		NRF_SPU->GPIOPORT[0].PERM = 0;
-	}
+#if defined(CONFIG_NRF_GPIO0_PIN_MASK_SECURE)
+	NRF_SPU->GPIOPORT[0].PERM = CONFIG_NRF_GPIO0_PIN_MASK_SECURE;
+#endif
 
-#ifdef CONFIG_SOC_NRF5340_CPUAPP
-	if (IS_ENABLED(CONFIG_SPM_NRF_P1_NS)) {
-		/* Configure GPIO pins to be Non-Secure */
-		NRF_SPU->GPIOPORT[1].PERM = 0;
-	}
+#if defined(CONFIG_NRF_GPIO1_PIN_MASK_SECURE)
+	NRF_SPU->GPIOPORT[1].PERM = CONFIG_NRF_GPIO1_PIN_MASK_SECURE;
```

Comment:
AS commented above, you should use HAL calls here as well.

Why did you change from `IS_ENABLED` ?

Reply:
The change is from:
 ```
 #ifdef CONFIG_SOC_NRF5340_CPUAPP
     if (IS_ENABLED(CONFIG_SPM_NRF_P1_NS)) {
 ```
 to
 ```
 #if defined(CONFIG_NRF_GPIO1_PIN_MASK_SECURE)
 ```
 Can't use `CONFIG_NRF_GPIO1_PIN_MASK_SECURE` in an IS_ENABLED when it is not defined.